dae/dae/gene_scores/gene_scores.py

```diff
@@ -158,20 +158,28 @@ def get_genes(
         self, score_id: str,
         score_min: float | None = None,
         score_max: float | None = None,
+        values: list[str] | None = None,
     ) -> set[str]:
-        """Return set of genes for a score between a min and max value."""
+        """Return set of genes for
+        a score between a min and max value or
+        genes with certain gene score values."""
         score_value_df = self.get_score_df(score_id)
         df = score_value_df[score_id]
-        if score_min is None:
-            score_min = float("-inf")
-        if score_max is None:
-            score_max = float("inf")
-
-        index = np.logical_and(
-            df.to_numpy() >= score_min,
-            df.to_numpy() <= score_max)
-        index = np.logical_and(index, df.notna())
-        genes = score_value_df[index].gene
+        if values is None:
+            if score_min is None:
+                score_min = float("-inf")
+            if score_max is None:
+                score_max = float("inf")
+
+            index = np.logical_and(
+                df.to_numpy() >= score_min,
+                df.to_numpy() <= score_max)
+            index = np.logical_and(index, df.notna())
+            genes = score_value_df[index].gene
+        else:
+            genes = score_value_df.loc[
+                score_value_df["gene-score"].isin([float(v) for v in values])
```

Comment:
Why are values being cast to float if we are dealing with a categorical gene score (according to the commit message)?

Reply:
For now gene score values are numerical. But the frontend uses universally strings because in the future both string and numerical values will be supported. A conversion is needed when searching strings.